graphics/generate-asciinema-gif

```diff
@@ -36,34 +37,33 @@ spoonfeed() {
 
 function script() {
   spoonfeed 2 'git ma~chete dis~cover --ch~ecked-out-since="2 weeks ago"'
-  spoonfeed 3 'y^'
-  spoonfeed 3 '~clear'
+  spoonfeed 5 'y^'
+  spoonfeed 1 '~clear'
 
   spoonfeed 1 'git ma~chete st~atus --l~ist-commits'
-  spoonfeed 4 '~clear'
+  spoonfeed 5 '~clear'
 
   spoonfeed 1 'git chec~kout d~evelop'
   spoonfeed 1 'git ma~chete tr~averse'
-  spoonfeed 3 'y^'
-  spoonfeed 3 'yq^'
-  spoonfeed 2 '~clear'
+  spoonfeed 4 'y^'
+  spoonfeed 4 'yq^'
+  spoonfeed 3 '~clear'
 
   spoonfeed 1 'git ma~chete st~atus --l~ist-commits'
-  spoonfeed 4 '~clear'
+  spoonfeed 5 '~clear'
 
   spoonfeed 1 'git ma~chete tr~averse'
-  spoonfeed 3 'y^'
-  spoonfeed 3 'yq^'
+  spoonfeed 4 'y^'
+  spoonfeed 2 'yq^'
   spoonfeed 5 '~clear'
   spoonfeed 1 '~exit'
 }
 
 init_contents='set -e -o pipefail -u; PS1=\(\\\$\(date\ +%H:%M:%S\)\)\ \$\ ; export GIT_MACHETE_DIM_AS_GRAY=true; export GIT_SEQUENCE_EDITOR=:'
 nested_command="bash --init-file <(echo '$init_contents')"
 script | asciinema rec --command="bash -c \"$nested_command\"" "$cast"
-# `head -n -3` to remove the 3 final lines responsible for the terminating `exit`
-# Note: for some reason, the resulting gif manifested weird behavior for widths lower than 120;
-# I needed to crop the right margin in an external tool.
-asciicast2gif -s 1 -S 1 -w 120 -h 35 <(head -n -3 "$cast") "$gif"
-echo "$gif"
+# `head -n ...` to remove the 3 final lines responsible for the terminating `exit`
+agg --speed 0.75 --cols 140 --rows 40 --font-size 12 <(head -n $(( $(wc -l < "$cast") - 2 )) "$cast") "$gif_tmp"
+gifsicle --crop 0,0+890x600 --colors 32 "$gif_tmp" > "$gif"
```

Comment:
890x600 is still larger than the original image:

![image](https://user-images.githubusercontent.com/3383210/193901551-4089b7cb-ce1c-4afe-b7fa-7b65da54c25c.png)

TBD if this can be done purely via `agg` options, if not then pls add a comment in code as to why

Reply:
@PawelLipski I had to increase the gif size to match the text, i checked how it looks when inserted in README.md and the result is really good so i would stick with 890x600, wdyt?

Reply:
Huh okay... lemme see 🕵🏻

Reply:
>    b. having `--cols 140 -- rows 40` and then doing a `gifsicle --crop`
>    ❓

I uploaded the gif named `cols_120_rows_35_no_crop.gif` for you to see

> 3. also AFAICS there's no underline of the current branch, might be sth wrong with the terminal... do you see the underline of current branch in general in your terminal (outside of `asciinema` context)?

@PawelLipski What do you mean by `underline of the current branch`?

Reply:
> What do you mean by underline of the current branch?

See `develop` in this screenshot:

![image](https://user-images.githubusercontent.com/3383210/194129286-26373124-e051-4de0-b828-d7c5626c641c.png)

This was present in the old gif, can't see it in the new version :/

Reply:
@PawelLipski Hmm, are you sure that it's present in the old gif? I can't see it anywhere in this readme -> https://github.com/VirtusLab/git-machete

Reply:
@PawelLipski So i took a look at the branch underlying and it is present outside of `asciinema`. I tried changing `--colors` parameter and run it in different terminal but still got no underlaying, can you run it on your machine and see if it works for you?

Reply:
@PawelLipski Also we can always come back to the no longer supported `asciicast2gif` that produces the branch underlaying instead of `agg` to create the gif

Reply:
Huh I've tried on Linux, same problems :///

Okay, so if you could:
1. fall back to `asciicast2gif`
2. open an issue against `agg` with a minimal repro (just `asciinema rec --command='echo -e "\033[4mlolxd"'`)
3. leave the `agg` line in script commented out with a TODO pointing to that issue